examples/Cassie/cassie_rbt_state_estimator.h

```diff
@@ -10,24 +10,111 @@
 #include "systems/framework/output_vector.h"
 #include "systems/framework/timestamped_vector.h"
 #include "examples/Cassie/datatypes/cassie_out_t.h"
+#include "examples/Cassie/cassie_utils.h"
 
 namespace dairlib {
 namespace systems {
 
+using Eigen::Vector3d;
+using Eigen::VectorXd;
+using Eigen::MatrixXd;
+using Eigen::Isometry3d;
+using systems::OutputVector;
+
+using drake::systems::Context;
+using drake::systems::EventStatus;
+using drake::systems::DiscreteValues;
+using drake::systems::DiscreteStateIndex;
+
 /// Translates from a TimestamedVector of cassie torque commands into
 /// a cassie_user_in_t struct for transmission to the real robot.
 class CassieRbtStateEstimator : public drake::systems::LeafSystem<double> {
  public:
-  explicit CassieRbtStateEstimator(
-      const RigidBodyTree<double>&);
+  explicit CassieRbtStateEstimator(const RigidBodyTree<double>&,
+                                   bool is_floating_base);
+  void solveFourbarLinkage(double* left_heel_spring,
+                           double* right_heel_spring,
+                           const VectorXd& q_init) const;
+
  private:
-  void Output(const drake::systems::Context<double>& context,
-      OutputVector<double>* output) const;
+  void AssignImuValueToOutputVector(
+      OutputVector<double>* output, const cassie_out_t& cassie_out) const;
+  void AssignActuationFeedbackToOutputVector(
+      OutputVector<double>* output, const cassie_out_t& cassie_out) const;
+  void AssignNonFloatingBaseStateToOutputVector(
+      OutputVector<double>* output, const cassie_out_t& cassie_out) const;
+  void AssignFloatingBaseStateToOutputVector(
+      OutputVector<double>* output, const VectorXd& state_est) const;
+
+  EventStatus Update(const Context<double>& context,
+                     DiscreteValues<double>* discrete_state) const;
+
+  void CopyStateOut(const Context<double>& context,
+                    OutputVector<double>* output) const;
+
+  void contactEstimation(int* left_contact, int* right_contact,
+    DiscreteValues<double>* discrete_state,
+    const OutputVector<double>& output, const double& dt) const;
 
   const RigidBodyTree<double>& tree_;
-  std::map<std::string, int> positionIndexMap_;
-  std::map<std::string, int> velocityIndexMap_;
-  std::map<std::string, int> actuatorIndexMap_;
+  const bool is_floating_base_;
+
+  std::map<std::string, int> position_index_map_;
+  std::map<std::string, int> velocity_index_map_;
+  std::map<std::string, int> actuator_index_map_;
+
+  // Body indices
+  int left_thigh_ind_ = -1;
```

Comment:
Why assigned to -1?

Reply:
-1 was the value that indicates there is no such body name. I added it because I didn't know that the function `GetBodyIndexFromName` already returns -1 if the body name doesn't exist. I'll delete the initial value.